### PR TITLE
Apple Silicon build results blog post

### PR DESCRIPTION
**Pull request by daveverwer**
A few things to do before publishing:

- [x] Publish an up-to-date list of failing packages to gist and update the URL to the gist in the post.
- [x] Update build numbers, number of failing builds, and the percentage of failing builds.
- [x] Are there any new error messages?

### CODE REVIEW

File: Content/posts/building-3238-packages-for-apple-silicon.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+---
+date: 2020-08-20 12:00
+title: Building 3,238 Swift Packages for Apple Silicon
+description: As part of the Swift Package Index build system, we have processed what must be the most extensive test of Apple Silicon compatibility outside of Apple. Here's what we found.
+---
+
+As part of the [Swift Package Index build system](https://blog.swiftpackageindex.com/posts/launching-language-and-platform-package-compatibility/), we have processed what must be the most extensive test of Apple Silicon compatibility outside of Apple.
+
+At the time of writing, we currently have XX,XXX Apple Silicon builds in our database covering 3,238 packages. Why so many? We test up to [three versions of each package](https://swiftpackageindex.com/SwifQL/SwifQL/builds) and compile each with both `xcodebuild` and `swift build`.
```
Comment (daveverwer) [inline Content/posts/building-3238-packages-for-apple-silicon.md:9]: This number needs updating, do you have the query handy?

File: Content/posts/building-3238-packages-for-apple-silicon.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+I’m sure that many of the Apple Silicon DTK machines Apple has shipped around the world are working very hard, testing apps for compatibility, but I’m not sure any of them have worked quite as hard as ours has. 🚀
+
+## What did we find?
+
+First of all, and as you might expect, there’s plenty of good news. For packages with successful Intel builds, the *vast majority* are also successful when built for Apple Silicon. 👍
+
+There are failures though, and the most interesting are those where the build fails on Apple Silicon while the corresponding Intel build succeeds. In cases like this, that indicates a compatibility issue that will potentially need some attention from the package author. We’ve identified 70 packages (~2%) where this is the case, and have [compiled a list of them for reference](https://example.com/link/to/the/gist).
```
Comment (daveverwer) [inline Content/posts/building-3238-packages-for-apple-silicon.md:17]: Also update these numbers, if necessary.

Comment (daveverwer) [reply Content/posts/building-3238-packages-for-apple-silicon.md:17]: This line also includes the URL to the gist.

File: Content/posts/building-3238-packages-for-apple-silicon.md (hunk 1 above)
Comment (finestructure) [reply Content/posts/building-3238-packages-for-apple-silicon.md:9]:
```
select count(*)
from builds
where platform in ('macos-spm-arm', 'macos-xcodebuild-arm')
and status = 'ok'
```

8291 as of this morning

File: Content/posts/building-3238-packages-for-apple-silicon.md (hunk 2 above)
Comment (finestructure) [reply Content/posts/building-3238-packages-for-apple-silicon.md:17]: I had a copy paste error in my query (both SPM and xcodebuild) where showing SPM builds. Will need to take another look at result!

File: Content/posts/building-3238-packages-for-apple-silicon.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+There are a few common errors that we’re observing in these failures, and most of them are from one of the following problems:
+
+- `Float80` is unavailable.
+- `x86\_thread\_state64\_t` is unavailable.
+- `\_\_darwin\_arm\_thread\_state64' has no member '\_\_rsp'`.
+- `\_\_asm\_\_("movq %rdi, %r13")`.
+- `cannot convert value of type 'CVaListPointer' to expected argument type '\_\_darwin\_va\_list?'`.
+- Sometimes, the compiler crashes!
```
Comment (finestructure) [inline Content/posts/building-3238-packages-for-apple-silicon.md:26]: The underscore escapes seem to be redundant in this Markdown version:

Screenshot 2020-09-09 at 07 29 54

File: Content/posts/building-3238-packages-for-apple-silicon.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+As part of the [Swift Package Index build system](https://blog.swiftpackageindex.com/posts/launching-language-and-platform-package-compatibility/), we have processed what must be the most extensive test of Apple Silicon compatibility outside of Apple.
+
+At the time of writing, we currently have XX,XXX Apple Silicon builds in our database covering 3,238 packages. Why so many? We test up to [three versions of each package](https://swiftpackageindex.com/SwifQL/SwifQL/builds) and compile each with both `xcodebuild` and `swift build`.
+
+I’m sure that many of the Apple Silicon DTK machines Apple has shipped around the world are working very hard, testing apps for compatibility, but I’m not sure any of them have worked quite as hard as ours has. 🚀
+
+## What did we find?
+
+First of all, and as you might expect, there’s plenty of good news. For packages with successful Intel builds, the *vast majority* are also successful when built for Apple Silicon. 👍
```
Comment (finestructure) [inline Content/posts/building-3238-packages-for-apple-silicon.md:15]: "the _vast majority_ also successfully build for Apple Silicon"?

File: Content/posts/building-3238-packages-for-apple-silicon.md (hunk 3 above)
Comment (finestructure) [reply Content/posts/building-3238-packages-for-apple-silicon.md:26]: Also, I was a but terse when I compiled that list :D So for the fourth point we probably want to say

- inline assembly like `__asm__("movq %rdi, %r13")` fails

File: Content/posts/building-3238-packages-for-apple-silicon.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+Even though 70 failing packages might sound like a lot, it’s not that bad, especially when you consider that 45 of the failures all fail due to an issue in a base dependency, [`PerfectLib`](https://swiftpackageindex.com/PerfectlySoft/Perfect).
+
+## Where do we go from here?
+
+Even though the results are generally positive, it does show that there’s a bit of work that the community need to do to ensure the transition to Apple Silicon is as smooth as possible. 
+
+If you’re a package author and do not have a DTK machine at your disposal, please feel free to use ours via the Swift Package Index build system! Make changes to your packages, push them to a pre-release tag (for example, X.X.X-arm-beta-X) and the build system will pick up that tag and build it. You should see new build results, usually within a couple of hours of pushing a tag.
+
+If you have any problems with the build system, we’re available in the #swiftpackageindex channel over at the [SwiftPM Slack](https://swift-package-manager.herokuapp.com).
```
Comment (finestructure) [inline Content/posts/building-3238-packages-for-apple-silicon.md:36]: also perhaps "or raise an issue (<link>)"?

File: Content/posts/building-3238-packages-for-apple-silicon.md (hunk 3 above)
Comment (daveverwer) [reply Content/posts/building-3238-packages-for-apple-silicon.md:26]: Fixed

File: Content/posts/building-3238-packages-for-apple-silicon.md (hunk 4 above)
Comment (daveverwer) [reply Content/posts/building-3238-packages-for-apple-silicon.md:15]: Fixed

File: Content/posts/building-3238-packages-for-apple-silicon.md (hunk 5 above)
Comment (daveverwer) [reply Content/posts/building-3238-packages-for-apple-silicon.md]: Fixed.

File: Content/posts/building-3238-packages-for-apple-silicon.md (hunk 1 above)
Comment (daveverwer) [reply Content/posts/building-3238-packages-for-apple-silicon.md]: Fixed. I put in 12,942 as the total number of builds as this query only lists successful builds, and at this point we're talking about total.

**Comment by daveverwer** [comment]
This is ready to go @finestructure. If you could review and merge it in the morning that'd be fab! Thanks 👍

**Review by finestructure — APPROVED** [review]

**Pull request merged**